File: daggermvp/src/main/java/be/vergauwen/simon/daggermvp/ui/TestActivity.java
package be.vergauwen.simon.daggermvp.ui;

import android.os.Bundle;
import android.util.Log;
import be.vergauwen.simon.daggermvp.MVPApplication;
import be.vergauwen.simon.himura.MVPDaggerActivity;

public class TestActivity extends MVPDaggerActivity<TestPresenter, TestComponent>
    implements TestContract.View {

    private final static String TAG = TestActivity.class.getName();

    private boolean thingsShown = false;

    @Override
    protected void createComponent() {
        component = DaggerTestComponent.builder()
            .applicationComponent(MVPApplication.getComponent())
            .build();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.e(TAG, "onCreate()");
    }

    @Override
    protected void onResume() {
        super.onResume();
        presenter.loadThings();
    }

    @Override
    protected void onDestroy() {
        Log.e(TAG, "onDestroy()");
        super.onDestroy();

    }

    @Override
    public void showThings() {
        Log.e(TAG, ".showThings()");
        thingsShown = true;
    }

    public boolean isThingsShown() {
        return thingsShown;
    }

    public TestPresenter getPresenter() {
        return presenter;
    }
}

File: kotlin-dagger-mvp/src/main/kotlin/be/vergauwen/simon/kotlindaggermvp/ui/TestActivity.kt
package be.vergauwen.simon.kotlindaggermvp.ui

import android.os.Bundle
import android.util.Log
import be.vergauwen.simon.himurakotlin.MVPDaggerActivity
import be.vergauwen.simon.kotlindaggermvp.MVPApplication

class TestActivity : MVPDaggerActivity<TestContract.View, TestPresenter, TestComponent>(), TestContract.View {

  var isThingsShown = false
    private set

  override fun createComponent(): TestComponent =
      DaggerTestComponent.builder().applicationComponent(MVPApplication.component).build()

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    Log.e("TestActivity", "onCreate()")
  }

  override fun onResume() {
    super.onResume()
    presenter.loadThings()
  }

  override fun onDestroy() {
    Log.e("TestActivity", "onDestroy()")
    super.onDestroy()

  }

  override fun showThings() {
    Log.e("TestActivity", ".showThings()")
    isThingsShown = true
  }

  fun presenterLoadThings() {
    presenter.loadThings()
  }

  fun testPresenterNotNull(): Boolean {
    return presenter != null;
  }
}